driver-kotlin-coroutine/build.gradle.kts

```diff
@@ -65,6 +65,7 @@ dependencies {
 
     api(project(path = ":bson", configuration = "default"))
     api(project(path = ":driver-reactive-streams", configuration = "default"))
+    implementation(project(path = ":bson-kotlin", configuration = "default"))
```

Comment:
Adds a direct dependency - should we also add kotlinx here as well?

Reply:
With just bson-kotlin:

Pros: Minimizes the dependencies - doesn't drag in kotlinx serialization.
Cons: bson-kotlinx becomes an extra opt in requirement.

```
implementation("org.mongodb:mongodb-driver-kotlin-coroutine:4.10.0")
implementation("org.mongodb:bson-kotlinx:4.10.0")
```

I think this PR is the best course of action - the automatic extra dependency on bson-kotlin is small. So don't think we should add kotlinx automagically.

Reply:
Yeah, I'm just not sure how widely kotlinx serialization will be used.  I guess we can always make it non-optional in the future.  We may need a documentation update for this.